utils/Query.ts

```diff
@@ -45,45 +45,63 @@ async function getBot(id: string, topLevel=true):Promise<Bot> {
 		.where({ id })
 		.orWhere({ vanity: id, trusted: true })
 		.orWhere({ vanity: id, partnered: true })
-	if (res[0]) {
-		const discordBot = await DiscordBot.users.fetch(res[0].id).then(r=> r).catch(() => null) as DiscordUser
-		if(!discordBot) return null
-		const botMember = await getMainGuild()?.members?.fetch(res[0].id).catch(e=> e) as GuildMember
-		res[0].flags = res[0].flags | (discordBot.flags?.bitfield & DiscordUserFlags.VERIFIED_BOT ? BotFlags.verified : 0) | (res[0].trusted ? BotFlags.trusted : 0) | (res[0].partnered ? BotFlags.partnered : 0)
-		res[0].tag = discordBot.discriminator
-		res[0].avatar = discordBot.avatar
-		res[0].name = discordBot.username
-		res[0].category = JSON.parse(res[0].category)
-		res[0].owners = JSON.parse(res[0].owners)
-		if(botMember) {
-			if(discordBot.flags.has(UserFlags.BotHTTPInteractions)) {
-				res[0].status = 'online'
-			} else if(!botMember.presence) {
-				res[0].status = 'offline'
-			} else {
-				res[0].status = botMember.presence.activities.some(r => r.type === ActivityType.Streaming) ? 'streaming' : botMember.presence.status
-			}
-		} else {
-			res[0].status = null
-		}
-		delete res[0].trusted
-		delete res[0].partnered
-		if (topLevel) {
-			res[0].owners = await Promise.all(
-				res[0].owners.map(async (u: string) => await get._rawUser.load(u))
-			)
-			res[0].owners = res[0].owners.filter((el: User | null) => el).map((row: User) => ({ ...row }))
-		}
 
-		await knex('bots').update({ name: discordBot.username }).where({ id })
+	if (!botRow[0]) return null
+
+	// FIXME: Handle this properly
```

Comment:
you may assert; further checks are being considered to be implemented in near future.